extensions/load-balancer/gce-extension/src/main/java/org/apache/stratos/gce/extension/GCELoadBalancer.java

```diff
@@ -0,0 +1,391 @@
+        else {
+            //first we need to take the port range. So arrange ipList in ascending order
+            Collections.sort(ipList);
+            //take the first one and last one
+            portRange = Integer.toString(ipList.get(0)) + "-" + Integer.toString(ipList.get(ipList.size() - 1));
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Port range set as: " + portRange);
+        }
+
+        //create the forwarding rule
+        gceOperations.createForwardingRule(gceClusterConfigurationHolder.getForwardingRuleName(),
+                gceClusterConfigurationHolder.getTargetPoolName(), PROTOCOL, portRange);
+        log.info("Created configuration for cluster");
+    }
+
+    @Override
+    public void start() throws LoadBalancerExtensionException {
+        //Configuration has completed
+        log.info("GCE Load balancer instance started");
+
+    }
+
+    @Override
+    public void stop() throws LoadBalancerExtensionException {
+
+        log.info("GCE Load Balancer is stopping");
+
+        //iterate through hash map and remove all
+        Iterator iterator = clusterToLoadBalancerConfigurationMap.entrySet().iterator();
+        while (iterator.hasNext()) { //for each configuration
+            Map.Entry clusterIDLoadBalancerConfigurationPair = (Map.Entry) iterator.next();
+            deleteConfigurationForCluster((String) clusterIDLoadBalancerConfigurationPair.getKey());
+        }
+
+        log.info("GCE Load balancer stopped");
+    }
+
+    /**
+     * @throws LoadBalancerExtensionException
+     */
+    @Override
+    public void reload() throws LoadBalancerExtensionException {
+
+        //nothing to do here
+        log.info("Configuration reloaded");
+
+    }
+
+    /**
+     * Create a valid target pool name
+     *
+     * @param clusterId - Id of the cluster
+     * @return - a proper name for target pool
+     */
+    private String targetPoolNameCreator(String clusterId) {
+        //create a valid target pool name by using cluster ID
+        //remove spaces, make all to lower case, replace all "." --> "-"
+        //add name prefix
+        String targetPoolName = GCEContext.getInstance().getNamePrefix().toLowerCase() + "-" +
+                clusterId.trim().toLowerCase().replace(".", "-");
+        //length should be les than 62 characters
+        if (targetPoolName.length() >= 62) {
+            targetPoolName = targetPoolName.substring(0, 62);
+        }
+        return targetPoolName;
+    }
+
+    /**
+     * Create a valid forwarding rule name
+     *
+     * @param clusterID - Id of the cluster
+     * @return - a proper name for forwarding rule
+     */
+    private String forwardingRuleNameCreator(String clusterID) {
+        String forwardingRuleName = GCEContext.getInstance().getNamePrefix()
+                + "-fr-" + clusterID.trim().toLowerCase().replace(".", "-");
+        //length should be les than 62 characters
+        if (forwardingRuleName.length() >= 62) {
```

Comment:
The value 62 might need to be moved to a constant and explain why this limit is used.